packages/@react-spectrum/checkbox/src/Checkbox.tsx

```diff
@@ -68,6 +71,9 @@ function Checkbox(props: SpectrumCheckboxProps, ref: FocusableRef<HTMLLabelEleme
         console.warn(`${key} is unsupported on individual <Checkbox> elements within a <CheckboxGroup>. Please apply these props to the group instead.`);
       }
     }
+    if (!props.value) {
```

Comment:
Do an explicit null check in case the value is an empty string or something

```suggestion
    if (props.value == null) {
```